src/modules/job-exec/exec.c

```diff
@@ -45,6 +45,7 @@ struct exec_ctx {
     const char * mock_exception;   /* fake exception */
```

Comment:
commit message for 9652288a830450c8979be49eb279fcebfa587b8d:

I'm not quite parsing this (can an in-progress barrier also be complete?)
> - a shell exits while an in-progress barrier is complete

Reply:
Sorry, that must be a typo (I don't understand it either!)

Reply:
Fixed to `a shell exits before an in-progress barrier is complete`